controllers/users/users-schema.js

```diff
@@ -0,0 +1,14 @@
+import mongoose from 'mongoose';
+
+const UsersSchema = mongoose.Schema({
+    username: {type: String, required: true},
+    firstName: String,
+    lastName: String,
+    type: {type: String, enum: ['REG USER', 'PRO CHIEF', 'RECIPE CREATOR'], default: 'REG USER',
```

Comment:
Tested on Postman and I'm able to update the user to a type that is not listed in the enum. For example, I can update my user to have a type of NONE

Reply:
When the user sign up or edit their profile from the front end, they choose their user type from the dropdown list, if they choose nothing, their default user type will be REG USER. So they will not have a chance to type in NONE as the user type. 

I tried to use a class object to replace the enum class in ts and use that object (UsersType for example) instead of String as type, but in js schema does not take a self-defined object as type. If we use "type: Object", it works but will lead to the similar error you mentioned here, as we can use any types of Object for the user type.